client/my-sites/domains/domain-management/components/primary-domain-selector/index.tsx

```diff
@@ -0,0 +1,217 @@
+	};
+
+	const onSubmit = ( event: FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+
+		const domain = domains.find( ( d ) => d.domain === selectedDomain );
+		if ( ! domain || ! selectedDomain ) {
+			return;
+		}
+
+		setIsSettingPrimaryDomain( true );
+		onSetPrimaryDomain( selectedDomain, () => setIsSettingPrimaryDomain( false ), domain.type );
+	};
+
+	const trackUpgradeClick = ( isPlanUpgrade: boolean = false ) => {
+		recordTracksEvent( 'calypso_primary_site_address_nudge_cta_click', {
+			cta_name: isPlanUpgrade ? 'buy_a_plan' : 'add_custom_domain',
+		} );
+	};
+
+	if ( validPrimaryDomains.length > 0 && ! selectedDomain ) {
+		setSelectedDomain( validPrimaryDomains[ 0 ].domain );
+	}
+
+	return (
+		<Card className="domains-set-primary-address">
+			<CardHeading
+				className="domains-set-primary-address__title"
+				isBold={ true }
+				size={ 16 }
+				tagName="h2"
+			>
+				{ translate( 'Primary site address' ) }
+			</CardHeading>
+			<>
+				<p className="domains-set-primary-address__subtitle">
+					{ translate(
+						'The current primary address set for this site is: {{strong}}%(selectedDomain)s{{/strong}}.',
+						{
+							args: { selectedDomain: primaryDomain as string },
+							components: {
+								strong: <strong />,
+							},
+						}
+					) }{ ' ' }
+					{ ! canUserSetPrimaryDomainOnThisSite &&
+						translate(
+							"Your site plan doesn't allow to set a custom domain as a primary site address. {{planUpgradeLink}}Upgrade your plan{{/planUpgradeLink}}.",
+							{
+								components: {
+									planUpgradeLink: (
+										<a
+											href={ '/plans/' + primaryDomain }
+											onClick={ () => trackUpgradeClick( true ) }
+										/>
+									),
+								},
+							}
+						) }
+					{ validPrimaryDomains.length > 1 &&
+						canUserSetPrimaryDomainOnThisSite &&
+						translate(
+							'You can change it by selecting a different address from the list below. {{learnMoreLink}}Learn more{{/learnMoreLink}}.',
+							{
+								components: {
+									learnMoreLink: (
+										<InlineSupportLink supportContext="primary-site-address" showIcon={ false } />
+									),
+								},
+							}
+						) }
+					{ validPrimaryDomains.length < 1 &&
+						canUserSetPrimaryDomainOnThisSite &&
+						translate(
+							'Before changing your primary site address you must {{domainSearchLink}}register{{/domainSearchLink}} or {{mapDomainLink}}connect{{/mapDomainLink}} a new custom domain.',
+							{
+								components: {
+									domainSearchLink: (
+										<a
+											href={ '/domains/add/use-my-domain/' + primaryDomain }
+											onClick={ () => trackUpgradeClick( false ) }
+										/>
+									),
+									mapDomainLink: (
+										<a
+											href={ '/domains/add/' + primaryDomain }
+											onClick={ () => trackUpgradeClick( false ) }
+										/>
+									),
+								},
+							}
+						) }
+				</p>
+				{ validPrimaryDomains.length > 1 && (
```

Comment:
```suggestion
				{ validPrimaryDomains.length >= 1 && (
```
Same as the last comment, I think this should be `>= 1`